vllm_v1/sample/sampler.py

```diff
@@ -0,0 +1,161 @@
+"""A layer that samples the next tokens from the model's outputs."""
+from dataclasses import dataclass
+from math import inf
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+
+from vllm_v1.sample.metadata import SamplingMetadata
+from vllm_v1.outputs import SamplerOutput
+
+_SAMPLING_EPS = 1e-5
+
+
+class Sampler(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(
+        self,
+        logits: torch.Tensor,
+        sampling_metadata: SamplingMetadata,
+    ) -> SamplerOutput:
+        logits = self.apply_temperature(logits, sampling_metadata.temperature)
+        logits = self.apply_top_k_top_p(logits, sampling_metadata)
```

Comment:
Yeah I think we should have a good support for that. At the moment, I didn't have a chance to consider it.

Reply:
Maybe add a TODO then? This can be added quickly add this after merging this PR. Meanwhile pluggable logits processor should be documented in the feature set as well cc @simon-mo

Reply:
Sg. Added a TODO.